OpenRA.Game/Server/Server.cs

```diff
@@ -387,35 +387,53 @@ void SetOrderLag()
 			SyncLobbyGlobalSettings();
 		}
 
-		void DispatchOrdersToClient(Connection c, int client, int frame, byte[] data)
+		void Dispatch(Connection connTo, Connection connFrom, ServerOrder order, int frame = 0, byte[] data = null)
 		{
-			try
-			{
-				SendData(c.Socket, BitConverter.GetBytes(data.Length + 4));
-				SendData(c.Socket, BitConverter.GetBytes(client));
-				SendData(c.Socket, BitConverter.GetBytes(frame));
-				SendData(c.Socket, data);
-			}
-			catch (Exception e)
+			data = data ?? order.Serialize();
+
+			var from = connFrom != null ? connFrom.PlayerIndex : 0;
+			var targetConnections = connTo != null ? new List<Connection> { connTo } :
+				(connFrom != null ? Conns.Except(connFrom) : Conns).ToList();
+
+			foreach (var c in targetConnections)
 			{
-				DropClient(c);
-				Log.Write("server", "Dropping client {0} because dispatching orders failed: {1}", client.ToString(), e);
+				try
+				{
+					SendData(c.Socket, BitConverter.GetBytes(data.Length + 4));
+					SendData(c.Socket, BitConverter.GetBytes(from));
+					SendData(c.Socket, BitConverter.GetBytes(frame));
+					SendData(c.Socket, data);
+				}
+				catch (Exception e)
+				{
+					DropClient(c);
+					Log.Write("server", "Dropping client {0} because dispatching orders failed: {1}", c, e);
```

Comment:
What is the first parameter now the connection object? Will that produce a useful string?